packages/slice-machine/src/features/customTypes/customTypesBuilder/CustomTypeProvider.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+  Dispatch,
+  ReactNode,
+  SetStateAction,
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { CustomType } from "@prismicio/types-internal/lib/customtypes";
+
+import useSliceMachineActions from "@src/modules/useSliceMachineActions";
+import {
+  AutoSaveStatusType,
+  useAutoSave,
+} from "@src/features/autoSave/useAutoSave";
+import { getFormat } from "@src/domain/customType";
+import { updateCustomType } from "@src/apiClient";
+
+import { CUSTOM_TYPES_MESSAGES } from "../customTypesMessages";
+
+type CustomTypeContext = {
+  customType: CustomType;
+  autoSaveStatus: AutoSaveStatusType;
```

Comment:
For me, I was planning to have 2 providers. As the provider, don't have the save logic since it's the hook and only contain things purely related to custom types or slices.

There are some duplications, but I would say healthy one that help us maintain the project without risking crashing everything.

You can see what I'm planning to do here: https://github.com/prismicio/slice-machine/blob/xru/auto-save-slice-builder/packages/slice-machine/src/features/slices/sliceBuilder/SliceBuilderProvider.tsx
And you can see that there are difference related to slices.

So for me 2 providers that reuse the hook yes.

Reply:
Hum, they will never be combined, since it's display from within the builder you never have both in the same view.